### DISCUSSION TITLE
TSV: `-f tsv -t commonmark_x`: table header, align columns `|`, alignment of data in particular columns

### DISCUSSION THREAD

**Discussion by lamyergeier** [jgm/pandoc#9769] [Q&A]
I have following questions about, rendering commonmark_x from TSV

## Header

1. Can I set that the input data is without a header  (so make the "table header", in the output, empty)?
2. Possible to set table header in the command, if not available in the Input Data?

## Alignment

1. Can the output render all columns in a straightline?
2. Can we define particular column numbers to have data in that column as right, left or center aligned?

# Sample Data

```
$ flatpak list -d  --runtime | pandoc -f tsv -t commonmark_x 
```

```markdown
| Fedora Platform | Shared libraries | org.fedoraproject.Platform | 39 | f39 | x86_64 | fedora | system | org.fedoraproject.Platform/x86_64/f39 | c77a6da6d276 | \- | 4.5 GB | system,alt-id=75ff9532aa82 |
|----|----|----|----|----|----|----|----|----|----|----|----|----|
| Fedora Platform | Shared libraries | org.fedoraproject.Platform | 40 | f40 | x86_64 | fedora | system | org.fedoraproject.Platform/x86_64/f40 | 49985e4ca19b | \- | 4.7 GB | system,alt-id=1202e767a1fd |
| Calf | LV2 plug-in suite | org.freedesktop.LinuxAudio.Plugins.Calf | 0.90.3 | 23.08 | x86_64 | flathub | system | org.freedesktop.LinuxAudio.Plugins.Calf/x86_64/23.08 | 8c6f837ef7c9 | \- | 16.6 MB | system |
| LSP | Linux Studio Plugins LADSPA/LV2/VST/VST3/CLAP | org.freedesktop.LinuxAudio.Plugins.LSP | 1.2.15 | 23.08 | x86_64 | flathub | system | org.freedesktop.LinuxAudio.Plugins.LSP/x86_64/23.08 | 48d3e1d09484 | \- | 72.8 MB | system |
```

**Comment by jgm** [comment]
> Can I set that the input data is without a header (so make the "table header", in the output, empty)?

Currently the code has
```hs
             toHeaderRow l = [toRow l | not (null l)]
```
but I don't think it's actually possible to create a null row in CSV (a row with no cells).  So this was the intended method, but it doesn't work.  I suppose we could change this so that if the first row contains only empty cells, we get a header-less table?  An approach that could work currently is to use a Lua filter to rearrange things.

> Possible to set table header in the command, if not available in the Input Data?

You could use a Lua filter to insert a header.

> Can the output render all columns in a straightline?

We used to do this in every case, but we recently changed things so that the alignment only happens if the rows are <= COLUMNS.  Trying to align very long lines just produces a mess.  You can use `--columns=5000` or something and that might have the desired effect.

> Can we define particular column numbers to have data in that column as right, left or center aligned?

You can adjust these things with a Lua filter, but there's no way to control it using the TSV reader itself.
